### ISSUE TITLE
Why is prefix slower than append?

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#550]
It's odd that `prefix` is so (relatively) slow here:

``` factor
IN: scratchpad : foo ( -- ) "bcd" CHAR: a prefix drop ;
IN: scratchpad gc [ 1,000,000 [ foo ] times ] time
Running time: 0.179098138 seconds

IN: scratchpad : bar ( -- ) "a" "bcd" append drop ;
IN: scratchpad gc [ 1,000,000 [ bar ] times ] time
Running time: 0.094251288 seconds

IN: scratchpad : baz ( -- ) CHAR: a 1string "bcd" append drop ;
IN: scratchpad gc [ 1,000,000 [ baz ] times ] time
Running time: 0.121946706 seconds
```

**Comment by AlexIljin** [comment]
Why is this issue still open? Was the patch rejected?
It's not clear what to do next here.

**Comment by mrjbq7** [comment]
There are some downsides to specializing, which basically instead of having a single generic code path have specialized paths that are jumped into if the types of the input match the specialized ones.  That makes it faster and makes the generated code bigger.

I'll try and look again at this patch and see if the performance gains can be verified and see if I still like the patch.

**Comment by mrjbq7** [comment]
Fixed in eddacb35a3c26fea0b88172fadd24aef1835efa1.
